File: source/aoBlogs2/Addons/BlogClass.vb
Imports Contensive.Addons.Blog.Controllers
Imports Contensive.Addons.Blog.Models
Imports Contensive.BaseClasses

Namespace Views
    Public Class BlogClass
        Inherits AddonBaseClass
        '
        '=====================================================================================
        ''' <summary>
        ''' Blog Addon
        ''' </summary>
        ''' <param name="CP"></param>
        ''' <returns></returns>
        Public Overrides Function Execute(ByVal CP As CPBaseClass) As Object
            Try
                '
                ' -- requests model - todo the controller for each view should handle its own requests
                Dim blogBodyRequest As New BlogBodyRequestModel(CP)
                Dim legacyRequest = New View.RequestModel(CP)
                '
                ' -- get blog settings
                Dim blog As BlogModel = BlogModel.verifyBlog(CP, blogBodyRequest.instanceGuid)
                If (blog Is Nothing) Then Return "<!-- Could not find or create blog from instanceId [" & blogBodyRequest.instanceGuid & "] -->"
                '
                ' -- process view requests
                Select Case blogBodyRequest.srcViewId
                    Case Else
                        '
                        ' -- default process nothing
                End Select
                '
                ' -- create next view
                Select Case blogBodyRequest.dstViewId
                    Case Else
                        '
                        ' -- default view
                End Select
                '
                Dim app As New ApplicationEnvironmentModel(CP, blog, blogBodyRequest.entryId)
                '
                ' -- get legacy Blog Body -- the body is the area down the middle that includes the Blog View (Article View, List View, Edit View)
                Dim legacyBlogBody As String = BlogBodyView.getBlogBody(CP, app, legacyRequest, blogBodyRequest)
                '
                ' -- sidebar wrapper
                Dim blogSidebarHtml As String = SidebarView.getSidebarView(CP, app, legacyRequest, legacyBlogBody)
                '
                ' -- if editing enabled, add the link and wrapperwrapper
                Return genericController.addEditWrapper(CP, blogSidebarHtml, blog.id, blog.name, BlogModel.tableMetadata.tableMetadata.contentName)
                '
            Catch ex As Exception
                CP.Site.ErrorReport(ex, "execute")
                Return "<p>The blog is not currently available.</p>"
            End Try
        End Function

    End Class
    '
    Public Class BlogBodyRequestModel
        Public Property instanceGuid As String
        Public Property srcViewId As Integer
        Public Property dstViewId As Integer
        Public Property entryId As Integer
        '
        Public Sub New(cp As CPBaseClass)
            srcViewId = cp.Doc.GetInteger(rnFormID)
            instanceGuid = cp.Doc.GetText("instanceId")
            If (String.IsNullOrWhiteSpace(instanceGuid)) Then instanceGuid = "BlogWithoutinstanceId-PageId-" & cp.Doc.PageId
            entryId = cp.Doc.GetInteger(RequestNameBlogEntryID)
        End Sub

    End Class
End Namespace

File: source/aoBlogs2/Addons/BlogClass.cs
using System;
using Contensive.Addons.Blog.Controllers;
using Contensive.Addons.Blog.Models;
using Contensive.BaseClasses;

namespace Contensive.Addons.Blog.Views {
    public class BlogClass : AddonBaseClass {
        // 
        // =====================================================================================
        /// <summary>
        /// Blog Addon
        /// </summary>
        /// <param name="CP"></param>
        /// <returns></returns>
        public override object Execute(CPBaseClass CP) {
            try {
                // 
                // -- requests model - todo the controller for each view should handle its own requests
                var blogBodyRequest = new BlogBodyRequestModel(CP);
                var legacyRequest = new Models.View.RequestModel(CP);
                // 
                // -- get blog settings
                var blog = BlogModel.verifyBlog(CP, blogBodyRequest.instanceGuid);
                if (blog is null)
                    return "<!-- Could not find or create blog from instanceId [" + blogBodyRequest.instanceGuid + "] -->";
                // 
                // -- process view requests
                switch (blogBodyRequest.srcViewId) {
                    default: {
                            break;
                        }
                        // 
                        // -- default process nothing
                }
                // 
                // -- create next view
                switch (blogBodyRequest.dstViewId) {
                    default: {
                            break;
                        }
                        // 
                        // -- default view
                }
                // 
                var app = new ApplicationEnvironmentModel(CP, blog, blogBodyRequest.entryId);
                // 
                // -- get legacy Blog Body -- the body is the area down the middle that includes the Blog View (Article View, List View, Edit View)
                string legacyBlogBody = BlogBodyView.getBlogBody(CP, app, legacyRequest, blogBodyRequest);
                // 
                // -- sidebar wrapper
                string blogSidebarHtml = SidebarView.getSidebarView(CP, app, legacyRequest, legacyBlogBody);
                // 
                // -- if editing enabled, add the link and wrapperwrapper
                return DesignBlockBase.Controllers.DesignBlockController.addDesignBlockEditWrapper(CP, blogSidebarHtml, blog, Models.BlogModel.tableMetadata.contentName);
                //return genericController.addEditWrapper(CP, blogSidebarHtml, blog.id, blog.name, BlogModel.tableMetadata.contentName);
            }
            // 
            catch (Exception ex) {
                CP.Site.ErrorReport(ex, "execute");
                return "<p>The blog is not currently available.</p>";
            }
        }

    }
    // 
    public class BlogBodyRequestModel {
        public string instanceGuid { get; set; }
        public int srcViewId { get; set; }
        public int dstViewId { get; set; }
        public int entryId { get; set; }
        // 
        public BlogBodyRequestModel(CPBaseClass cp) {
            srcViewId = cp.Doc.GetInteger(constants.rnFormID);
            instanceGuid = cp.Doc.GetText("instanceId");
            if (string.IsNullOrWhiteSpace(instanceGuid))
                instanceGuid = "BlogWithoutinstanceId-PageId-" + cp.Doc.PageId;
            entryId = cp.Doc.GetInteger(constants.RequestNameBlogEntryID);
        }

    }
}